optaplanner-website-root/content/blog/2023/02/21/OptaPlanner-9-is-coming.adoc

```diff
@@ -0,0 +1,104 @@
+
+From now on and until further notice, the OptaPlanner project will be releasing two binary streams:
+
+* OptaPlanner 8.x as you know and love it.
+* OptaPlanner 9.x adapted to the new world.
+
+OptaPlanner 9.x will be our latest and greatest, but there will be no difference between these two streams, unless strictly necessary for the switch to the new Jakarta APIs.
+We are not removing any deprecated APIs, we are making no functional changes.
+
+The two streams will also have the same minor version, to indicate that they belong together.
+For this reason, the first release of the 9.x stream won't be OptaPlanner 9.0.0.Final.
+Instead it will (likely) be 9.37.0.Final, to go along with 8.37.0.Final.
+
+
+== How do I choose?
+
+It is simple, and we've created a flow chart to help you:
+
+image::migration-flowchart.png[Which OptaPlanner stream to choose?]
+
+To summarize, switch to OptaPlanner 9:
+
+* If you're starting a new project,
+* or if you don't know what Quarkus and Spring Boot are,
+* or if you don't use them.
+
+Otherwise stay on OptaPlanner 8, but be advised:
+OptaPlanner 8 will not be around forever, and you should plan for migration.
+Maintaining two streams presents a considerable effort for our team,
+and we will only do it until Quarkus 3 and Spring Boot 3 gain enough traction.
+It may take 6 months, it may take a year, but most likely not longer.
+
+
+== How do I migrate to OptaPlanner 9?
+
+Everything in `optaplanner-core` works exactly as it has before.
+For the vast majority of our users, we believe that switching to OptaPlanner 9 will require just a single line change in your Maven POM file.
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.optaplanner</groupId>
+    <artifactId>optaplanner-bom</artifactId>
+    <version>9.34.0.Beta1</version> <!-- Used to say 8.33.0.Final -->
+    <type>pom</type>
+    <scope>import</scope>
+</dependency>
+----
+
+For users of `optaplanner-quarkus` or `optaplanner-spring-boot-starter`, the changes will be larger:
+
+* All imports of `javax.*` packages will need to be replaced by their `jakarta.*` equivalents, including changes to dependencies.
+* Dependencies on Quarkus 2 need to be changed to Quarkus 3.
+* Dependencies on Spring Boot 2 need to be changed to Spring Boot 3. In case of Spring Boot 3, a migration to Java 17 is also necessary. That is the decision of the Spring framework team, and we can not influence that.
+
+However, we have created tooling to simplify this transition to a point where it is also a single line of code. Simply run the following command while in the root directory of your project:
+
+[source,shell]
+----
+rewrite:run \
+  -Drewrite.configLocation="${optaplanner_file}" \
+  -Drewrite.recipeArtifactCoordinates=org.optaplanner:optaplanner-migration:"$project_version" \
+  -Drewrite.exclusions=optaplanner-operator/**,optaplanner-examples/data/** \
+  -Drewrite.activeRecipes=org.optaplanner.openrewrite.Quarkus3 \
+  -Dfull \
+  -Dquickly \
+  -Dmigration
+----
```

Comment:
@rsynek Please provide the correct command.

Reply:
I think something along these lines could work after merging https://github.com/kiegroup/optaplanner/pull/2635 :

```
curl -o ToLatest9.yml https://raw.githubusercontent.com/kiegroup/optaplanner/main/optaplanner-migration/src/main/resources/META-INF/rewrite/ToLatest9.yml
mvn clean org.openrewrite.maven:rewrite-maven-plugin:LATEST:run \
   -Drewrite.configLocation=ToLatest9.yml \
   -Drewrite.recipeArtifactCoordinates=org.optaplanner:optaplanner-migration:8.34.0.Final \
   -Drewrite.activeRecipes=org.optaplanner.migration.ToLatest9
```